### PR TITLE
removed old TargetFrameworks 

### LINKED ISSUE #172: Time to modernize csproj and drop old TargetFramework entries
Looks like we need to make some updates to the CSPROJ. It's surprising that `sl5` is still a supported target in there! Time to modernize :-)

_Originally posted by @nblumhardt in https://github.com/sprache/Sprache/issues/162#issuecomment-857194276_

### PR DESCRIPTION
**Pull request by MagicAndre1981**
removed old TargetFrameworks ( sl5;net35;net40;net45;netstandard1.0 ) from csproj. Fixes #172

### CODE REVIEW

File: src/Sprache/Sprache.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <AssemblyTitle>Sprache</AssemblyTitle>
     <VersionPrefix>2.3.2</VersionPrefix>
```
Comment (nblumhardt) [inline src/Sprache/Sprache.csproj]: Breaking change -> bump to 3.0.0?

File: src/Sprache/Sprache.csproj [hunk 2]
```diff
@@ -18,57 +18,4 @@
...
-  <FrameworkPathOverride>C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v3.5\Profile\Client</FrameworkPathOverride>
-  </PropertyGroup>
-
-  <PropertyGroup Condition=" '$(TargetFramework)' == 'net40' ">
-    <DefineConstants>$(DefineConstants);STRING_JOIN_ENUMERABLE</DefineConstants>
-    <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
-  </PropertyGroup>
-
-  <PropertyGroup Condition=" '$(TargetFramework)' == 'net45' ">
-    <DefineConstants>$(DefineConstants);STRING_IS_ENUMERABLE;STRING_JOIN_ENUMERABLE</DefineConstants>
```
Comment (nblumhardt) [inline src/Sprache/Sprache.csproj:60]: We need to add `<DefineConstants>$(DefineConstants);STRING_IS_ENUMERABLE;STRING_JOIN_ENUMERABLE</DefineConstants>` to the root properties (these enable features that are supported on all the newer targets).

File: src/Sprache/Sprache.csproj (hunk 1 above)
Comment (MagicAndre1981) [reply src/Sprache/Sprache.csproj]: do this later on your own

File: src/Sprache/Sprache.csproj (hunk 2 above)
Comment (MagicAndre1981) [reply src/Sprache/Sprache.csproj:60]: done

**Pull request merged**